packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition.js

```diff
@@ -99,6 +102,14 @@ Blockly.Blocks.trade_definition = {
         };
     },
     onchange(event) {
+        if (event.type === 'ui' && !this.isInit) {
+            this.isInit = true;
+            initErrorHandlingListener('keydown');
+        }
+        if (Blockly.selected === null && this.isInit) {
+            this.isInit = false;
+            removeErrorHandlingEventListener('keydown');
+        }
```

Comment:
`this.isInit` is being set to `true` in the first condition, are we expecting to still run the second condition after meeting the first one? If there are no major complication and the 2nd condition `(Blockly.selected === null && this.isInit` is okay to be initiated then we can leave it as is, otherwise we should make it `else if` instead.

Reply:
yes we should make it as `else if` because we don't intend to run the other one if one gets executed will do this change 🙇